source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/update/models.ts

```diff
@@ -0,0 +1,49 @@
+import { fromApiOrganization, fromApiPerson, IContactSearchResult } from '@/interfaces';
+import { Api_PropertyContact } from '@/models/api/Property';
+import { stringToNull } from '@/utils/formUtils';
+
+export class PropertyContactFormModel {
+  public id: number = 0;
+  public propertyId: number = 0;
+  public contact: IContactSearchResult | undefined = undefined;
+  public primaryContactId: string = '';
+  public purposeDescription: string = '';
+  public isDisabled: boolean = false;
+  public rowVersion: number = 0;
+
+  public toApi(): Api_PropertyContact {
+    return {
+      id: this.id,
+      propertyId: this.propertyId,
+      organizationId: this.contact?.organizationId ?? null,
+      organization: null,
+      personId: this.contact?.personId ?? null,
+      person: null,
+      primaryContactId: this.primaryContactId !== '' ? Number(this.primaryContactId) : null,
+      primaryContact: null,
+      purpose: stringToNull(this.purposeDescription),
+      isDisabled: this.isDisabled,
+      rowVersion: this.rowVersion,
+    };
+  }
+
+  static fromApi(model: Api_PropertyContact | null): PropertyContactFormModel {
+    const newFormModel = new PropertyContactFormModel();
+    newFormModel.id = model?.id || 0;
+    newFormModel.propertyId = model?.propertyId || 0;
+    newFormModel.isDisabled = model?.isDisabled || false;
+    newFormModel.rowVersion = model?.rowVersion || 0;
+    if (model?.organization) {
+      newFormModel.contact = fromApiOrganization(model.organization);
+    } else if (model?.person) {
```

Comment:
I'm not sure if this is safe, right now it seems the api is returning the organization of a person whenever a person is selected.

Reply:
fair point

Reply:
I checked and this should be fine given the model is a property contact, and that can only be either a Person or organization but not both.